File: for-local-test/python/389.find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        sum = 0
        for c in t:
            sum += ord(c)
        for c in s:
            sum -= ord(c)
        return chr(sum)

if __name__ == "__main__":
    solution = Solution()
    s = "abcd"
    t = "abcde"
    result = solution.findTheDifference(s, t)
    print(f"The extra character in \"{t}\" compared to \"{s}\" is: '{result}'")

File: for-local-test/389.find-the-difference.java

class Solution {
    public char findTheDifference(String s, String t) {
        int sum = 0;
        for (char c : t.toCharArray()) {
            sum += c;
        }
        for (char c : s.toCharArray()) {
            sum -= c;
        }
        return (char) sum;
    }
    public static void main(String[] args){
        Solution sol = new Solution();
        System.out.println(sol.findTheDifference("abcd", "abcde")); // Output: 'e'
    }
}


